vapi/namespace/simulator/simulator.go

```diff
@@ -240,3 +245,143 @@ func (h *Handler) getService(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 	}
 }
+
+var namespacesMap = make(map[string]*namespace.NamespacesInstanceInfo)
+
+func (h *Handler) namespaces(w http.ResponseWriter, r *http.Request) {
+	subpath := r.URL.Path[len(internal.NamespacesPath):]
+	subpath = strings.Replace(subpath, "/", "", -1)
+
+	switch r.Method {
+	case http.MethodGet:
+		if len(subpath) > 0 {
+			if result, contains := namespacesMap[subpath]; contains {
+				vapi.StatusOK(w, result)
+			} else {
+				vapi.ApiErrorNotFound(w)
+			}
+			return
+		} else {
+			result := make([]namespace.NamespacesInstanceSummary, 0, len(namespacesMap))
+
+			for k, v := range namespacesMap {
+				entry := namespace.NamespacesInstanceSummary{
+					ClusterId:    v.ClusterId,
+					Namespace:    k,
+					ConfigStatus: v.ConfigStatus,
+					Description:  v.Description,
+					Stats:        v.Stats,
+				}
+				result = append(result, entry)
+			}
+
+			vapi.StatusOK(w, result)
+		}
+	case http.MethodPatch:
+		if len(subpath) > 0 {
+			if entry, contains := namespacesMap[subpath]; contains {
+				var spec namespace.NamespacesInstanceUpdateSpec
+				vapi.Decode(r, w, &spec)
+
+				entry.VmServiceSpec = spec.VmServiceSpec
+
+				vapi.StatusOK(w)
+				return
+			}
+		}
+
+		vapi.ApiErrorNotFound(w)
+	case http.MethodPost:
+		var spec namespace.NamespacesInstanceCreateSpec
+		vapi.Decode(r, w, &spec)
```

Comment:
Similar to above comment, or just:
```go
if !vapi.Decode(r, w, &spec) {
    return
}
```

Reply:
Thanks, there're were a few other places where I applied this idea